brave/src/main/java/brave/internal/propagation/InjectorFactory.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.internal.propagation;
+
+import brave.Request;
+import brave.propagation.Propagation;
+import brave.propagation.Propagation.RemoteSetter;
+import brave.propagation.Propagation.Setter;
+import brave.propagation.TraceContext;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+public final class InjectorFactory {
```

Comment:
This is an internal type exctracted from B3. I don't plan to do a huge amount of polishing on this in this PR as I'm at a severe time deficit. I will puts with it post merge though and any significant changes needed pre-merge

Reply:
I lied.. polish is here now